datafusion-examples/examples/composed_extension_codec.rs

```diff
@@ -71,8 +71,10 @@ async fn main() {
 
     // deserialize proto back to execution plan
     let runtime = ctx.runtime_env();
+    let state = ctx.state();
+    let config_options = state.config_options();
     let result_exec_plan: Arc<dyn ExecutionPlan> = proto
-        .try_into_physical_plan(&ctx, runtime.deref(), &composed_codec)
+        .try_into_physical_plan(&ctx, config_options, runtime.deref(), &composed_codec)
```

Comment:
This is an API change, but I think it is required to thread the config options through as each argument is specific

We could potentially improve the `try_into_physical_plan` API (as a follow on PR) to make it easier to update the API in the future using a trait or something like

https://github.com/apache/datafusion/blob/e99e02b9b9093ceb0c13a2dd32a2a89beba47930/datafusion/expr/src/expr_schema.rs#L39-L38




So this would look something like 
```rust
pub trait ProtobufContext { 
  /// return a function registry
  fn function_registry(&self) -> &dyn FunctionRegistry;
  /// return the runtime env
  fn runtime_env(&self) -> &RuntimeEnv;
  /// return the config options
  fn config_options(&self) -> &ConfigOptions;
  /// return extension codec
  fn extension_codec(&self) -> &dyn PhysicalExtensionCodec;
}
```

```rust
impl AsExecutionPlan for protobuf::PhysicalPlanNode {
...
    fn try_into_physical_plan(
        &self,
        registry: &dyn FunctionRegistry,
        config_options: &ConfigOptions,
        runtime: &RuntimeEnv,
        extension_codec: &dyn PhysicalExtensionCodec,
    ) -> Result<Arc<dyn ExecutionPlan>> {